### PR TITLE
Implement `ClaimAuctionLotBidder` contract

### PR DESCRIPTION
**Pull request by errfrom**

### CODE REVIEW

File: src/Contract/StartBidding.purs [hunk 1]
```diff
@@ -215,28 +216,18 @@ mkStartBiddingContractWithErrors (StartBiddingContractParams params) = do
...
-      <> mkAuctionToken standingBidTokenName
-
-  mkAuctionToken :: TokenName -> Value
-  mkAuctionToken tokenName =
-    Value.singleton auctionInfo.auctionId tokenName one
+  auctionCs :: CurrencySymbol
+  auctionCs = auctionInfo.auctionId
+
+  worker :: TransactionOutput -> Boolean
+  worker (TransactionOutput txOut) =
```
Comment (euonymos) [inline src/Contract/StartBidding.purs]: I guess the name `worker` here might have been more self-descriptive.

File: src/Contract/ClaimAuctionLotBidder.purs [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+  -- Get seller pkh:
+  sellerPkh <- wrap <$> toPubKeyHash auctionTermsRec.sellerAddress
+    ?? ClaimAuctionLotBidder_Error_CouldNotGetSellerPkh
+
+  let
+    validatorHashes = unwrap $ validatorHash <$> validators
+
+    mkAuctionToken :: TokenName -> Value
+    mkAuctionToken tokenName = Value.singleton auctionCs tokenName one
```
Comment (euonymos) [inline src/Contract/ClaimAuctionLotBidder.purs:140]: Don't we have something like `foo = \cs tn -> Value.singleton cs tn one`?

File: src/Contract/ClaimAuctionLotBidder.purs [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+      , Lookups.validator (unwrap validators).auctionEscrow
+      , Lookups.validator (unwrap validators).standingBid
+      ]
+
+  lift $ submitTxReturningContractResult {} $ emptySubmitTxData
+    { lookups = lookups
+    , constraints = constraints
+    }
+
+queryStandingBidUtxo :: AuctionInfo -> Contract (Maybe (Utxo /\ StandingBidState))
```
Comment (euonymos) [inline src/Contract/ClaimAuctionLotBidder.purs]: We already have `queryStandingBidUtxo` with the same signature, though I see its logic is different. But for clearness' sake, we might want to find another name (I not sure, I will re-read both functions more carefully).

**Review by euonymos — APPROVED** [review]
I love the way you write the contracts, they are really neat and easy to read. I suggest improving the naming in two places, though it's up to you @errfrom.

File: src/Contract/StartBidding.purs (hunk 1 above)
Comment (errfrom) [reply src/Contract/StartBidding.purs]: Agreed, renamed to `isValidAuctionEscrowUtxo`.

File: src/Contract/ClaimAuctionLotBidder.purs (hunk 2 above)
Comment (errfrom) [reply src/Contract/ClaimAuctionLotBidder.purs:140]: I think using `Value.singleton` is the "canonical" way to build a single-token Value.

File: src/Contract/ClaimAuctionLotBidder.purs (hunk 3 above)
Comment (errfrom) [reply src/Contract/ClaimAuctionLotBidder.purs]: Very good catch! Extracted utxo queries into a separate module `HydraAuctionOffchain.Contract.QueryUtxo` and eliminated the mentioned duplication.

**Review by euonymos — APPROVED** [review]

**Pull request merged**